ynput/ayon-core
# Max: updated implementation of save_scene + small QOL improvements to host

## Changelog Description
Ported version of: https://github.com/ynput/OpenPype/pull/6186

## Testing notes:
1. Start 3dsmax
2. Check that on the right hand side it has defaulted to the 'create' view/menu/mode
3. Open the 'Publish/Create' AYON Menu
4. Create new 'workfile' publish
5. Publish
6. Check the log of save_scene is NOT saving if no changes to scene were made; or is saving when changes have been made (it should display INFO or DEBUG information in either case).
